include/deal.II/base/segment_data_out.h

```diff
@@ -0,0 +1,160 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2025 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+#ifndef dealii_segment_data_out_h
+#define dealii_segment_data_out_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/data_out_base.h>
+#include <deal.II/base/point.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class generates graphical output for line segment objects.
+ *
+ * Line segments are one-dimensional objects embedded in a @p dim dimensional
+ * space. This class creates output data that can be visualized as lines in
+ * visualization tools. Each segment is represented as a pair of Point<dim>
+ * objects defining the start and end points.
+ */
+template <int dim>
```

Comment:
This should be relatively easy: Would you mind renaming `dim -> spacedim`? I think that would better match the general use of these names in the library where here, you are using `dim==1` objects in a `spacedim` dimensional space.